src/media/local-audio-track.ts

```diff
@@ -0,0 +1,28 @@
+import { LocalTrack } from './local-track';
+
+export type MicrophoneConstraints = Pick<
+  MediaTrackConstraints,
+  | 'deviceId'
+  | 'echoCancellation'
+  | 'autoGainControl'
+  | 'noiseSuppression'
+  | 'sampleRate'
+  | 'sampleSize'
+  | 'channelCount'
+>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TrackEffect = any;
+/**
+ * Represents a audio track for a audio source.
+ */
+export class LocalAudioTrack extends LocalTrack {
+  /**
+   * Makes sure to apply the encoderConfig for the audio.
+   *
+   * @param encoderConfig - Encoder config for audio.
+   */
+  setEncoderConfig(encoderConfig: MicrophoneConstraints): void {
```

Comment:
not sure about the name - the function name and argument use "EncoderConfig" name, but the type is "Constraints", we should be consistent.

Reply:
ya previouly we had a different type now its different